### PR TITLE
feat(meetings): add sendDtmf feature

### PR DESCRIPTION
**Pull request by adamweeks**
# Pull Request Template

## Description

Added a `meeting.sendDtmf` feature to the meetings plugin and kitchen sink samples.

Fixes # [https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-117172](SPARK-117172)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Plugin Meetings

**Test Configuration**:
* SDK Version
* Node/Browser Version
* NPM Version

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 1]
```diff
@@ -2742,4 +2742,29 @@ export default class Meeting extends StatelessWebexPlugin {
   unlockMeeting() {
     return MeetingUtil.unlockMeeting(this.inMeetingActions, this.meetingRequest, this.locusUrl);
   }
+
+  /**
+   * Sends DTMF tones to the current meeting
+   * @param {String} tones
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: As a developer, I'm not sure what `tones` could be, can we create a shape/constants list of possible tones or a small documentation update to describe what possible tones can be used as this is marked a public method?

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 2]
```diff
@@ -2742,4 +2742,29 @@ export default class Meeting extends StatelessWebexPlugin {
...
+   * @returns {Promise}
+   * @public
+   * @memberof Meeting
+   */
+  sendDtmf(tones) {
+    let errorMessage;
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: A suggestion on this, not a requirement: Create a class-member-local method to generate the error instead:
```js
const reject = (message) => {
  LoggerProxy.logger.error(message);
  return Promise.reject(new Error(message);
}

if(condition) { return reject('message'); }
```
@mwegman suggested something similar to this in the `services=>serviceHost` class methods and it worked well.

File: packages/node_modules/@webex/plugin-meetings/test/unit/spec/meeting/index.js [hunk 3]
```diff
@@ -588,6 +592,44 @@ describe('plugin-meetings', () => {
           });
         });
       });
+
+      describe('#sendDtmf', () => {
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/plugin-meetings/test/unit/spec/meeting/index.js]: Tests look good, they cover all possible outcomes inclusively and exclusively 👍

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 4]
```diff
@@ -2742,4 +2742,29 @@ export default class Meeting extends StatelessWebexPlugin {
...
+
+    if (this.locusInfo && this.locusInfo.self) {
+      if (this.locusInfo.self.enableDTMF) {
+        return this.webex.internal.locus.sendDtmf(this.locusInfo, tones);
```
Comment (chrisadubois) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: discussed live

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js (hunk 1 above)
Comment (adamweeks) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: Done, thanks for the feedback!

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js (hunk 4 above)
Comment (adamweeks) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: Now using meeting request object instead.

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js (hunk 2 above)
Comment (adamweeks) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: Nice solution! Implemented!

**Comment by adamweeks** [comment]
@InteractiveTimmy @arun3528 @chrisadubois thanks for your feedback. I've updated as requested. Please re-review at your convenience.

File: packages/node_modules/@webex/plugin-meetings/src/meeting/request.js [hunk 5]
```diff
@@ -1,20 +1,22 @@
+import uuid from 'uuid';
 import {StatelessWebexPlugin} from '@webex/webex-core';
 
 import LoggerProxy from '../common/logs/logger-proxy';
 import {
```
Comment (adamweeks) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/request.js]: Sorted all these imports.

**Comment by adamweeks** [comment]
@webex/js-sdk-widgets-core testing

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 6]
```diff
@@ -2742,4 +2742,34 @@ export default class Meeting extends StatelessWebexPlugin {
...
+            locusUrl: this.locusInfo.self.url,
+            deviceUrl: this.deviceUrl,
+            tones
+          });
+      }
+
+      return reject('Meeting:index#sendDtmf --> cannot send dtmf, meeting does not have dtmf enabled');
+    }
+
+    return reject('Meeting:index#sendDtmf --> cannot send dtmf, meeting does not have a locus. Have you joined?');
```
Comment (lalli-flores) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: Not sure if third-parties would be using this, but saying "doesn't locus" would make sense for the average Joe?
Also, do you mind capitalizing DTMF as it's an acronym? thanks

**Pull request merged**